packages/ingester/src/ingestible.ts

```diff
@@ -0,0 +1,34 @@
+import { ColorPaletteName } from "@withorbit/core";
+
+export type Ingestible = {
+  sources: IngestibleSource[];
+};
+
+/**
+ * @TJS-type string
+ * @TJS-pattern ^[0-9a-zA-Z_\-]{22}$
+ */
+export type IngestibleSourceIdentifier = string & {
+  __sourceIDOpaqueType: never;
+};
+
+export interface IngestibleSource {
+  // unique identifier representing a source which tasks might share, used for clustering; can be a URL or an arbitrary UUID
+  identifier: IngestibleSourceIdentifier;
+  // title that can be used when displaying the source.
+  title: string;
+  // prompts associated with the given source
+  prompts: IngestiblePrompt[];
```

Comment:
Two somewhat invasive suggestions; feel free to take or leave.

**Prompt -> TaskSpec**
I think it might be more parsimonious to replace `prompts` with `taskSpecs` here? I think `TaskSpec` is the general form of what you're pointing to here. It would let you avoid redefining QA prompts below and mapping between tasks and prompts. And once we define non-memory tasks (which I do intend to do!) I expect to want to ingest them from Markdown in a similar way.

A very simple task type I expect to add would simply display some content, and then provide some controls which let you adjust the subsequent interval. I might use this to show myself visual inspiration, or for tasks like "sing a minor sixth upward", etc. I can imagine wanting to ingest these from my Markdown notes by tagging any block with `#orbitTask`.

Feel free to YAGNI me.

**Provenance**
For that matter, the other properties seem mostly duplicative of `TaskProvenance`. Could this whole thing become?:

```
export interface IngestibleSource {
   // unique identifier representing a source which tasks might share, used for clustering; can be a URL or an arbitrary UUID
   identifier: IngestibleSourceIdentifier;

   specs: TaskSpec[];
   provenance: TaskProvenance;
}